### PR TITLE
Add check-smtp

### PR DESCRIPTION
**Pull request by shiimaxx**
This plugin can be used check behave for smtp server.
Plugin sends following commands in basis of SMTP Protocol.

- `HELO/EHLO`
- `STARTTLS` (optional)
- `AUTH` (optional, only `PLAIN` supported)
- `MAIL`
- `RSET`
- `QUIT`

### CODE REVIEW

File: check-smtp/lib/check-smtp.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+	Host     string  `short:"H" long:"host" default:"localhost" description:"Hostname"`
+	Port     string  `short:"p" long:"port" default:"25" description:"Port"`
+	FQDN     string  `short:"F" long:"fqdn" description:"FQDN used for HELO"`
+	SMTPS    bool    `short:"s" long:"smtps" description:"Use SMTP over TLS"`
+	StartTLS bool    `short:"S" long:"starttls" description:"Use STARTTLS"`
+	Auth     string  `short:"A" long:"authmech" description:"SMTP AUTH Authentication Mechanisms (only PLAIN supported)"`
+	User     string  `short:"U" long:"authuser" description:"SMTP AUTH username"`
+	Password string  `short:"P" long:"authpassword" description:"SMTP AUTH password"`
+	Warning  float64 `short:"w" long:"warning" default:"3" description:"Warning threshold (sec)"`
+	Critical float64 `short:"c" long:"critical" default:"5" description:"Critical threshold (sec)"`
```
Comment (itchyny) [inline check-smtp/lib/check-smtp.go]: I think it's not intuitive that the plugin has default thresholds. User should explicitly specify the monitoring thresholds. What do you think?

**Comment by itchyny** [comment]
Sorry for being late, we decided to accept the plugin.  I commented one concern on the default thresholds.Thank you for your contribution.

File: check-smtp/lib/check-smtp.go (hunk 1 above)
Comment (shiimaxx) [reply check-smtp/lib/check-smtp.go]: I think it is correct your concern.
For that reason deleted default value of options of threshold.

File: check-smtp/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  -H, --host=         Hostname (default: localhost)
+  -p, --port=         Port (default: 25)
+  -F, --fqdn=         FQDN used for HELO
+  -s, --smtps         Use SMTP over TLS
+  -S, --starttls      Use STARTTLS
+  -A, --authmech=     SMTP AUTH Authentication Mechanisms(only PLAIN supported)
+  -U, --authuser=     SMTP AUTH username
+  -P, --authpassword= SMTP AUTH password
+  -w, --warning=      Warning threshold (sec) (default: 3)
+  -c, --critical=     Critical threshold (sec) (default: 5)
```
Comment (itchyny) [inline check-smtp/README.md]: The default threshold is now removed so please update these lines.

Comment (shiimaxx) [reply check-smtp/README.md]: Sorry. I Updated it.

File: check-smtp/lib/check-smtp.go [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+	"net/smtp"
+
+	"fmt"
+
+	"crypto/tls"
+
+	"net"
+
+	"time"
```
Comment (itchyny) [inline check-smtp/lib/check-smtp.go]: I noticed too many new lines here.

File: check-smtp/lib/check-smtp.go [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+	elapsed := time.Since(stTime)
+
+	msg := fmt.Sprintf("%.3f seconds response time", elapsed.Seconds())
+
+	if opts.Critical != 0 && elapsed.Seconds() > opts.Critical {
+		return checkers.Critical(msg)
+	} else if opts.Warning != 0 && elapsed.Seconds() > opts.Warning {
+		return checkers.Warning(msg)
+	} else {
```
Comment (mattn) [inline check-smtp/lib/check-smtp.go]: In habitually, Go ｒecommend to ommit unnecessary "else".

File: check-smtp/lib/check-smtp.go (hunk 3 above)
Comment (shiimaxx) [reply check-smtp/lib/check-smtp.go]: Thanks, I fix import statement in https://github.com/mackerelio/go-check-plugins/pull/243/commits/30249cdb7cb9d033d93ba15e20b138bff8a96a16.

File: check-smtp/lib/check-smtp.go (hunk 4 above)
Comment (shiimaxx) [reply check-smtp/lib/check-smtp.go]: Thanks, I removed "else" following your advice.

**Comment by itchyny** [comment]
LGTM, Thank you.

**Pull request merged**